components/indicators/IndicatorVisualisation.js

```diff
@@ -310,8 +310,12 @@ const generateTrendTrace = (indicator, traces, goals, i18n) => {
 
     const highestDataYear = traces[0].y[traces[0].y.length - 1];
     const highestGoalYear = Math.max(
-      ...goals.map((goal) => goal.x[goal.x.length - 1])
+      ...goals.map((goal) => {
+        const goalYear = goal.x[goal.x.length - 1];
+        return goalYear ? parseInt(goalYear.split('-')[0], 10) : NaN;
```

Comment:
Instead of 
`parseInt(goalYear.split('-')[0], 10)`
you could maybe use
`(new Date(goalYear)).getFullYear()` to make the intention clearer.
Also, maybe rename goalYear variable to goalDate to be clearer.

Reply:
Done! thank you for suggestion @tituomin